x/ecocredit/client/tx.go

```diff
@@ -632,6 +645,15 @@ Parameters:
 					NewQuantity:       update.NewQuantity,
 					DisableAutoRetire: update.DisableAutoRetire,
 				}
+
+				// parse and set expiration
+				if update.NewExpiration != "" {
+					expiration, err := ParseDate("expiration", update.NewExpiration)
+					if err != nil {
+						return err
+					}
+					updates[i].NewExpiration = &expiration
+				}
```

Comment:
We would need to pass the whole order struct to make this work in which case only 2/3 of the code blocks could make use of the function because the update order struct uses a `NewExpiration` rather than `Expiration`.

The alternative would be to return the parsed date rather than setting it but I'm not sure how necessary it would be to extract in that case as the function would only add the condition checking if an empty string before parsing.

Reply:
In the proposal I suggested to use a pointer, so it's possible to do:
```
*dest = expiration
```

Reply:
I like the idea but the issue is passing `&updates[i].NewExpiration` as a pointer - `NewExpiration` is already a pointer. I still don't see how this would be possible aside from what I mentioned above.

Reply:
`updates[i].NewExpiration` is already a pointer - so no need to get a pointer of a pointer. Will commit in your branch.